olive/model/utils/path_utils.py

```diff
@@ -0,0 +1,36 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import logging
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+
+def normalize_path_suffix(file_or_dir_path: str, filename_with_suffix: str) -> str:
+    """Get the file full path.
+
+    The engine provides output paths to Olive passes that do not contain file extension
+    (these paths are generally locations in the cache). This function will convert such
+    paths to absolute file paths and also ensure the parent directories exist.
+    If the input path is already an ONNX file it is simply returned. Examples:
+
+    normalize_path_suffix("c:/foo/bar.onnx", "model.onnx") -> c:/foo/bar.onnx
```

Comment:
In which case we will have different file name here?

Reply:
For some of the passes, the output_model_path may contains the model_name which is defined by user.
![image](https://github.com/microsoft/Olive/assets/13343117/ca282341-a0d7-4bee-afd0-721e43bfad5a)

Reply:
![image](https://github.com/microsoft/Olive/assets/13343117/a945dbc5-0f59-4414-8a92-3c1ce19e74fc)